packages/block-library/src/navigation/edit/index.js

```diff
@@ -383,15 +371,19 @@ function Navigation( {
 	] );
 
 	const startWithEmptyMenu = useCallback( () => {
-		if ( navigationArea ) {
-			setAreaMenu( 0 );
-		}
-		setAttributes( {
-			ref: undefined,
+		registry.batch( () => {
```

Comment:
Why do we use `batch` here?

Reply:
I don't think it's necessarily useful here but basically I wanted to avoid rerenders which might cause "effects" to trigger before the whole function runs (all state updates happen). 

It can also be considered a perf improvement since it avoid multiple useless rerenders.